src/api/profile.ts

```diff
@@ -137,3 +137,72 @@ export const updatePersonalizedSchedule = async ({
     });
   });
 };
+
+export interface updateTalkShowStudioExperienceProps {
+  venueId: string;
+  userId: string;
+  experience: TalkShowStudioExperience;
+}
+
+export const updateTalkShowStudioExperience = async ({
+  venueId,
+  userId,
+  experience,
+}: updateTalkShowStudioExperienceProps) => {
+  const userProfileRef = getUserRef(userId);
+
+  const userData = (await userProfileRef.get()).data();
+
+  const newData = {
+    [`data.${venueId}`]: { ...userData?.data?.[venueId], ...experience },
+  };
+
+  userProfileRef.update(newData).catch((err) => {
+    Bugsnag.notify(err, (event) => {
+      event.addMetadata("context", {
+        location: "api/profile::updateTalkShowStudioExperience",
+        venueId,
+        userId,
+        event,
+        experience,
+      });
+
+      throw err;
+    });
+  });
+};
+
+export interface UpdateUserIdsProps {
+  venueId: string;
+  userId: string;
+  // TODO: unify updateUserIds, updateScreenShareStatus and updatePlaceInRoom methods into one
+  props: Record<string, string | number>;
+}
+
+export const updateUserIds = async ({
+  venueId,
+  userId,
+  props,
+}: UpdateUserIdsProps) => {
+  const userProfileRef = getUserRef(userId);
+
+  const userData = (await userProfileRef.get()).data();
+
+  const newData = {
+    [`data.${venueId}`]: { ...userData?.data[venueId], ...props },
+  };
```

Comment:
same, no fetching

Reply:
@mike-lvov Why do you say no fetching? I'm not sure if that is correct/ideal?